Source/OCMock/OCMockObject.m

```diff
@@ -261,50 +284,75 @@ - (void)forwardInvocation:(NSInvocation *)anInvocation
     }
     @catch(NSException *e)
     {
-        [exceptions addObject:e];
+        @synchronized(exceptions)
+        {
+            [exceptions addObject:e];
+        }
         [e raise];
     }
 }
 
 - (BOOL)handleInvocation:(NSInvocation *)anInvocation
 {
-    [invocations addObject:anInvocation];
-
+    @synchronized(invocations)
+    {
+        // When the method has a char* argument we do not retain the arguments. This makes it possible
+        // to match char* args literally and with anyPointer. Not retaining the argument means that
+        // in these cases tests that use their own autorelease pools may fail unexpectedly.
+        if(![anInvocation hasCharPointerArgument])
+            [anInvocation retainArguments];
+        [invocations addObject:anInvocation];
+    }
+    
```

Comment:
Hmm, that's possible.  =\  It's definitely necessary to retain the invocation arguments though, otherwise they won't be valid from thread-to-thread.  Really depending on how the autorelease scope goes they won't even be valid on the same thread in a lot of cases.  Something like this was already exploding.

```
@autoreleasepool {
    [mock stubbedMethodWithArgument:argument];
}
OCMVerifyAll(mock);
```